controllers/cloudkitty_controller.go

```diff
@@ -855,6 +859,53 @@ func (r *CloudKittyReconciler) generateServiceConfigs(
 		return err
 	}
 
+	prometheusEndpointSecret := &corev1.Secret{}
+	err = r.Client.Get(ctx, client.ObjectKey{
+		Name:      cloudkitty.PrometheusEndpointSecret,
+		Namespace: instance.Namespace,
+	}, prometheusEndpointSecret)
+	if err != nil {
+		Log.Info("Prometheus Endpoint Secret not found")
+	}
+	if instance.Spec.PrometheusHost == "" {
+		if prometheusEndpointSecret.Data != nil {
```

Comment:
Retrieve only the prometheus secret if PrometheusHost is empty. If the user chooses to inform the PrometheusHost, we dont really need to query the Kubernetes API to get the secret.